### PR TITLE
chore(all): switch to make and fix lint errors

### PR DESCRIPTION
**Pull request by technicallyty**
## What is the purpose of the change

removes magefiles for makefiles
fixes all linting errors, add nolint where applicable.

## Brief Changelog

theres a lot of changes here. most are renames, added error checks, small nits to appease the linter. 

ill list the main changes below:

## Moved/Renamed
- moved component_metadata -> ecs/component/metadata. this is due to linter not liking underscores in packages.
- test_utils is now testutils.
- IComponentMetaData -> ComponentMetadata

## Deleted
- deleted all magefiles

## Added
- added makefiles for our most common usages.

## Testing and Verifying

verified all current tests still work.

### CODE REVIEW

**Review by namrapatel — APPROVED** [review]
LGTM

**Comment by technicallyty** [comment]
note: waiting for @heronimus to wire up CI before merging

**Pull request merged**
